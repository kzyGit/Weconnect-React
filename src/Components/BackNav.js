import React, { Component } from 'react';
import '../Styles/App.css';
import { browserHistory } from 'react-router';

/**
 * Component that enables moving back to the previous path
 */
class BackNav extends Component {
  render() {
    return (
        <span className="glyphicon glyphicon-circle-arrow-left" id='backnav' onClick={browserHistory.goBack}>
        
        </span>
    );
  }
}
export default BackNav;