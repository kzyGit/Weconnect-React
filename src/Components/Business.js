import React, { Component } from 'react';
import '../Styles/App.css';
import { Base_url } from './Header';
import icon from '../Images/businessicon.png';
import axios from 'axios';
import swal from 'sweetalert';
import { browserHistory } from 'react-router';
import PropTypes from 'prop-types';

/**
 * This component renders a single business
 */
class Business extends Component {

  constructor(props) {
    super(props);
    this.state = {
      businesses: [],
      reviews: [],
      isHidden: true
    };
  }

  /**
   * Function to toggle display of 'Add reviews' button on user logged in or not
   */

  toggleHidden() {
    this.setState({
      isHidden: !this.state.isHidden
    });
  }

  /**
   * Querries the server to get a single business and its reviews
   * @returns {object} a single business object and a review object for the business
   */
  componentDidMount() {
    const bid = this.props.params.bid;
    axios.get(`${Base_url}/businesses/${this.props.params.bid}`).then(response => {
      this.setState({ businesses: response.data });
    }).catch(error => {
      if (error.response.status === 404) {
        const message = error.response.data.Error;
        swal("message!!", message, "error");
      }

    });

    axios.get(`${Base_url}/businesses/${bid}/review`).then(response => {
      this.setState({ reviews: response.data });
    }).catch(error => {
      if (error.response.status === 404) {
        document.getElementById("rev").innerHTML = ' * No reviews added yet';
      }
    });
  }

  render() {
    return (
      
        <div className="onebusinesscontent" style={{ padding: '25px' }}>
          <div style={{ textAlign: 'center' }}>
            <img src={icon} alt="Business Logo" style={{ height: '100px', width: '120px' }} className="img-thumbnail" /><br />
            <h4 id="details">{this.state.businesses.Name}</h4>
          </div>

          <div>
            <h4 id="details">About us</h4>
            <div>
              {this.state.businesses.description}
            </div>
          </div><br />

          <div>
            <h4 id="details">Location</h4>
            {this.state.businesses.Location}
          </div><br />

          <div>
            <h4 id="details">Contacts</h4>
            +254 721 988 543,+254 799 999 555, companyinfo@gmail.com
        </div><br />

          <div>
            <h4 id="details">Reviews
            {sessionStorage.loggedIn && <button onClick={this.toggleHidden.bind(this)} className="btn btn-primary" style={{ marginLeft: '10%' }}><span className="glyphicon glyphicon-plus-sign"></span> Add Review</button>}
            </h4>
            <span id='rev'></span>
            <div>
              {!this.state.isHidden &&
                <form onSubmit={this.addreview.bind(this)} id='reviewform'><br />
                  <textarea cols="30" name='content' className="form-control" style={{ width: '98%' }} required></textarea><br />
                  <button type="submit" className="btn btn-default">Submit</button><br /><br />
                </form>}
            </div>

            <div className="row"> <br />
              <div className="col-md-12" id="reviews" >

                {this.state.reviews.map(review =>
                  <div id="review" key={review.Id}>
                    {review.Review}<br />
                    <span id="reviewer"><span className="glyphicon glyphicon-user" style={{ paddingRight: '10px', color: 'black' }}></span> {review.createdBy}</span><br />
                  </div>)}
              </div>
            </div>
          </div>
        </div>
    );
  }

  /**
   * Makes api request to add a review for the business
   * @returns {string} success or error message
   */
  addreview = (e) => {
    e.preventDefault();

    const auth_token = sessionStorage.getItem("access_token");
    const config = { headers: { 'Authorization': "bearer " + auth_token } };
    const bid = this.props.params.bid;
    const content = e.target.elements.content.value;

    axios.post(`${Base_url}/businesses/${bid}/review`, {
      content: content
    }, config).then(response => {
      window.location.reload();
      swal({
        title: "Success!",
        text: response.data.Success,
        icon: "success"
      });
    })
      .catch(error => {
        if (error.response.status === 401) {
          const message = error.response.data.Error;
          swal("Error!!", message, "error");
          sessionStorage.removeItem('loggedIn');
          browserHistory.push('/login');
        }

        else if (error.response.status === 400) {
          const message = error.response.data.Error;
          swal("Error!!", message, "error");
        }
      });
  }
}

Business.propTypes = {
  params: PropTypes.object
};
export default Business;