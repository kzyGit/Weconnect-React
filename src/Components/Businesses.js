import React, { Component } from 'react';
import '../Styles/App.css';
import { Base_url } from './Header';
import { Link } from 'react-router';
import axios from 'axios';
import swal from 'sweetalert';
import JwPagination from 'jw-react-pagination';
import icon from '../Images/businessicon.png';

/**
 * This component renders all available businesses
 * @returns {obj} businesses
 * @param {string} e
 */
class Businesses extends Component {
  constructor() {
    super();
    this.onChangePage = this.onChangePage.bind(this);
    this.state = {
      pageOfItems: [],
      businesses: []
    };
  }

  /**
   * update local state with new page of items
   * @param {obj} pageOfItems
   * @returns {obj} businesses
   */
  onChangePage(pageOfItems) {
    this.setState({ pageOfItems });
  }
  componentDidMount() {
    axios.get(`${Base_url}/businesses`).then(response => {
      this.setState({ businesses: response.data });
    }).catch();
  }

  /**
   * Makes a server request to retrieve a business searched by name
   * @param {string} e
   * @return {object} businesses of the search name
   */
  searchByName = (e) => {
    e.preventDefault();

    const name = e.target.elements.searchname.value;

    axios.get(`${Base_url}/businesses?q=${name}`).then(res => {
      if (res.data.status_code === 204) {
        swal('Sorry, no business found');
        this.search.value = '';
      }
      else {
        this.setState({ businesses: res.data });
      }
    });
  }

  /**
   * Makes a server request to filter businesses of a specified location and category
   * @param {string} e
   * @return {object} businesses of a specified location and category
   */
  filter = (e) => {
    e.preventDefault();
    this.search.value = '';

    const category = e.target.elements.searchcategory.value;
    const location = e.target.elements.searchlocation.value;

    axios.get(`${Base_url}/businesses?category=${category}&location=${location}`).then(resp => {

      if (resp.data.status_code === 204) {
        swal('Sorry, no business found');
      }
      else {
        this.setState({ businesses: resp.data });
      }
    });
  }

  changename = (e) => {
    e.preventDefault();
    this.cat.value = '';
    this.loc.value = '';
  }

  changefilter = (e) => {
    e.preventDefault();
    this.search.value = '';
  }
  
  reloadpage = (e) => {
    e.preventDefault();
    window.location.reload();
  }

  render() {
    return (

      <div className="businessescontent">
        <div className="row">
        <h3 onClick={this.reloadpage} style={{ paddingLeft: '20px', cursor: 'pointer', color: 'maroon' }}>Businesses</h3><br />
          <div className="row">
          <div className="col-sm-8" id='filterspan'>
          
          <form onSubmit={this.filter} id="filterform">
              <span>Category: </span><span>
                <select name="searchcategory" ref={(ref) => this.cat = ref} onChange={this.changefilter} id="search" >
                  <option value="">All</option>
                  <option>Technology</option>
                  <option>Tourism & Hotels</option>
                  <option>Health</option>
                  <option>Education</option>
                  <option>Finance & Accounting</option>
                  <option>Farming</option>
                  <option>Ecommerse</option>
                  <option>Real Estate</option>
                  <option>Manufacturing</option>
                </select></span>

              <span style={{ paddingLeft:'20px' }}>Location: </span><span>
                <input type="text" ref={(ref) => this.loc = ref} onChange={this.changefilter} name="searchlocation" id="searchfilter"  /></span>
                <span><button type="submit" id="search" style={{ paddingRight: '20px' }}><span className="glyphicon glyphicon-search" ></span></button></span>

              </form>
            </div>
          <div className="col-sm-1"></div>
          <div className="col-sm-3">
          
          <form onSubmit={this.searchByName} id="searchform">
              <span><input type="text" onChange={this.changename} ref={(ref) => this.search = ref} id="searchname" name="searchname" placeholder="Search by name..." /></span>
              <span><button type="submit" id="search" style={{ paddingRight: '20px' }}><span className="glyphicon glyphicon-search" ></span></button></span>
            </form>
            
            </div>

          </div> <br /><br />
          <div className="col-md-12">
            {this.state.businesses.length > 0 &&
              <div>
                <div id='bcards'>
                    {this.state.businesses.length > 0 &&

                      this.state.pageOfItems.map(business =>

                        <div key={business.id} id='card' className="card col-md-3">
                        <div id="carditem">
                        
                          <h4 className="card-title">{business.name}</h4>
                          <div className="card-image">
                            <img src={icon} alt="Business Logo" style={{ height: '100px', width: '120px', marginTop:'5px' }} className="img-thumbnail" /><br />
                          </div><br />
                          <span id='label'>Category:</span> {business.category}<br /><br />
                          <span id='label'>Location:</span> {business.location}<br />
                          <br /><Link to={`/business/${business.id}`} id='businesslink'><button type="submit" className="btn btn-default">Read More</button></Link>
                          <br /><br /></div><br />
                        </div>
                      )}
                </div>

                <br /><br />

                <div style={{ float: 'right', textDecoration: 'none' }}>
                  <JwPagination items={this.state.businesses} pageSize={12} id="pagination" onChangePage={this.onChangePage} />
                </div>
              </div>
            }
            {this.state.businesses.length === 0 &&
              <img alt='' src="https://upload.wikimedia.org/wikipedia/commons/b/b1/Loading_icon.gif" id="loadpage" />
            }
          </div>
        </div>
      </div>
    );
  }
}

export default Businesses;
