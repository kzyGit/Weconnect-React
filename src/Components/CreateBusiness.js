import React, { Component } from 'react';
import '../Styles/App.css';
import Header, { Base_url } from './Header';
import Footer from './Footer';
import swal from 'sweetalert';
import axios from 'axios'
import { browserHistory } from 'react-router';

/**
 * This component handles creating a new business
 */
class CreateBusiness extends Component {

  state = {
    businesses: []
  };

  /**
   * Display content only when user is authorised. If not, redirect to login
   */
  componentDidMount() {

    if (!localStorage.loggedIn) {
      swal('Kindly login first to create a business !');
      browserHistory.push('/login')
    }
  }

  render() {
    return (
      <div className="row">
        <Header />
        <div className="col-md-3"></div>
        <div className="col-md-6" id="businesscontent">
          <h3>Create a business</h3><br />

          <form className="create-business-form" onSubmit={this.createBusiness}>
            <div className="form-group">
              <label>Name:</label>
              <input type="text" className="field" name='business_name' required />

            </div><br />
            <div className="form-group">
              <label>Location:</label>
              <input type="text" className="field" name='location' required />
            </div><br />

            <div className="form-group">
              <label>Category:</label>
              <select className="field" name='category' required>
                <option>Technology</option>
                <option>Tourism & Hotels</option>
                <option>Health</option>
                <option>Education</option>
                <option>Finance & Accounting</option>
                <option>Farming</option>
                <option>Ecommerse</option>
                <option>Real Estate</option>
                <option>Manufacturing</option>
              </select>
            </div><br />

            <div className="form-group">
              <label >About:</label><br /><br />
              <textarea id='textarea' rows='8' name='about' required></textarea>
            </div>
            <a style={{ float: 'right', marginTop: '180px' }}><button type="submit" className="btn btn-primary">Create</button></a>
          </form>

        </div>
        <div className="col-md-3"></div>
        <Footer />
      </div>
    );
  }

  /**
   * Makes a server request to post a new business
   * @return {object} a new business
   */
  createBusiness = (e) => {
    e.preventDefault()

    const business_name = e.target.elements.business_name.value;
    const location = e.target.elements.location.value;
    const category = e.target.elements.category.value;
    const about = e.target.elements.about.value;

    const access_token = localStorage.getItem("access_token")
    const config = {
      headers: { 'Authorization': "bearer " + access_token }
    }

    axios.post(`${Base_url}/businesses`, {
      business_name: business_name,
      location: location,
      category: category,
      about: about

    }, config).then(response => {
      browserHistory.push('/dashboard')
      swal({
        title: "Success!",
        text: response.data.Success,
        icon: "success",
        button: "OK",
      });
    })
      .catch(error => {

        if (error.response.status === 409) {
          const message = error.response.data.Error
          swal("Error!!", message, "error");
        }
        else if (error.response.status === 401) {
          const message = error.response.data.Error
          swal("Error!!", message, "error");
          localStorage.removeItem('loggedIn')
          browserHistory.push('/login')
        }
        else if (error.response.status === 400) {
          const message = error.response.data.Error
          swal("Error!!", message, "error");
        }
      });
  }
}
export default CreateBusiness;