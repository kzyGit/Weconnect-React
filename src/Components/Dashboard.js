import React, { Component } from 'react';
import '../Styles/App.css';
import { Base_url } from './Header';
import { Link } from 'react-router';
import axios from 'axios'
import swal from 'sweetalert';
import { browserHistory } from 'react-router';

/**
 * Display a users businesses and allows user to access edit and delete business features
 */
class Dashboard extends Component {

  state = {
    businesses: [],
    availablebusinesses: 'available'
  };

  /**
   * Shows businesses of the logged in user
   * @returns {object} state, businesses
   */
  componentDidMount() {

    if (!localStorage.loggedIn) {
      swal("Error!!", 'Login first to view your dashboard', "error");
      browserHistory.push('/login')
    }
    else {
      const auth_token = localStorage.getItem("access_token")
      const config = { headers: { 'Authorization': "bearer " + auth_token } }

      axios.get(`${Base_url}/mybusinesses`, config).then(response => {
        this.setState({ businesses: response.data });
      }).catch(error => {
        if (error.response.status === 401) {
          swal("Error!!", "Login session expired, Login again to view your dashboard", "error");

          localStorage.removeItem('loggedIn')
          browserHistory.push('/login')
        }
      });
    }
  }

  /**
   * Make a server request to delete a business
   * @param {int} business_id
   */
  deleteBusiness = (business_id) => {
    const auth_token = localStorage.getItem("access_token")
    const config = {
      headers: { 'Authorization': "bearer " + auth_token }
    }
    swal({
      text: "Are you sure you want to delete this business?",
      icon: "warning",
      buttons: true,
      dangerMode: true,
    })
      .then((willDelete) => {
        if (willDelete) {

          axios.delete(`${Base_url}/businesses/${business_id}`, config)
            .then(response => {
              this.componentDidMount()
              swal(response.data.Success, {
                icon: "success",
              });
              browserHistory.push('/dashboard')
            })
            .catch(error => {
              if (error.response.status === 404) {
                const message = error.response.data.Error
                swal("Error!!", message, "error");
              }
            })
        }
      });
  }

  render() {
    return (
      <div className="content">
        <div className="row">
          <h3 style={{ paddingLeft: '20px', color: 'maroon' }}>My Businesses</h3><br /><br />
          <div className="col-md-1"></div>
          <div className="col-md-10" >

            {this.state.businesses.length > 0 &&
              <table className="table">
                <thead>
                  <tr>
                    <th id="name-row">Business</th>
                    <th id="row">Category</th>
                    <th id="row">Location</th>
                    <th id="row">Edit</th>
                    <th id="row">Delete</th>
                  </tr>
                </thead>
                <tbody>

                  {this.state.businesses.map(business =>
                    <tr key={business.id}>
                      <td><Link to={`/business/${business.id}`} >{business.name}</Link></td>
                      <td>{business.category}</td>
                      <td>{business.location}</td>
                      <td>
                        <Link to={`/editBusiness/${business.id}`} ><button className="btn btn-primary">
                          <span className="glyphicon glyphicon-edit"></span> </button>
                        </Link>
                      </td>
                      <td id="deletebusiness"><button className="btn btn-danger" onClick={this.deleteBusiness.bind(this, business.id)}><span className="glyphicon glyphicon-trash"></span> </button></td>
                    </tr>
                  )}
                </tbody>
              </table>}

            <span id='rev'></span><br />
            Let your businesses join the WeConnect network
            <Link to="/addbusiness" style={{ paddingLeft: '30px', textDecoration: 'none' }}><span className="glyphicon glyphicon-hand-right"></span> Add Business</Link><br /><br /><br />
          </div>
          <div className="col-md-1"></div>
        </div>
      </div>
    );
  }
}

export default Dashboard;
