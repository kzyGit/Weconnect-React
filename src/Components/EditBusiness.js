import React, { Component } from 'react';
import '../Styles/App.css';
import { Base_url } from './Header';
import axios from 'axios';
import swal from 'sweetalert';
import { browserHistory } from 'react-router';
import PropTypes from 'prop-types';

/**
 * Component that updates a business
 */
class EditBusiness extends Component {

  constructor(props) {
    super(props);
    this.state = {
      businesses: [],
      name: '',
      location: '',
      category: '',
      id: '',
      about: ''
    };
    this.changelog = this.changelog.bind(this);
    this.editBusiness = this.editBusiness.bind(this);
  }

  /**
   * Make server request to get the business of the given id
   * @returns {object} a business
   */
  componentDidMount() {
    if (!localStorage.loggedIn) {
      swal("Error!!", 'Login first to update a business', "error");
      browserHistory.push('/home');
    }
    else {
      const auth_token = localStorage.getItem("access_token");
      const config = { headers: { 'Authorization': "bearer " + auth_token } };
      const bid = this.props.params.bid;

      axios.get(`${Base_url}/mybusinesses/${bid}`, config).then(response => {
        this.setState({
          businesses: response.data,
          id: response.data.Id,
          name: response.data.Name,
          location: response.data.Location,
          about: response.data.description,
          category: response.data.category
        });
      }).catch(error => {
        if (error.response.status === 404) {
          const message = error.response.data.Error;
          swal("message!!", message, "error");
        }
        else if (error.response.status === 401) {
          swal("Error!!", error.response.data.Error, "error");
          localStorage.removeItem('loggedIn');
          browserHistory.push('/login');
        }
      });
    }
  }

  /**
   * Make server request to update the business of the given id
   * @param {string} e
   * @returns {string} success message
   */
  editBusiness = (e) => {
    e.preventDefault();
    const { name, location, about, category } = this.state;
    const access_token = localStorage.getItem("access_token");
    const config = {
      headers: { 'Authorization': "bearer " + access_token }
    };
    const bid = this.props.params.bid;
    axios.put(`${Base_url}/businesses/${bid}`, {
      business_name: name,
      location: location,
      category: category,
      about: about

    }, config).then(response => {
      browserHistory.push('/dashboard');
      swal({
        title: "Success!",
        text: response.data.Success,
        icon: "success",
        button: "OK",
      });
    })
      .catch(error => {
        if (error.response.status === 409) {
          swal("Error!!", error.response.data.Error, "error");
        }
        else if (error.response.status === 401) {
          swal("Error!!", error.response.data.Error, "error");
          localStorage.removeItem('loggedIn');
          browserHistory.push('/login');
        }
        else if (error.response.status === 400) {
          swal("Error!!", error.response.data.Error, "error");
        }
      });
  }

  /**
   * update local state with new page of items
   * @param {string} e
   * @returns {state} changestate
   */
  changelog(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }
  render() {
    return (
      <div className="row">
        <div className="col-md-3"></div>
        <div className="col-md-6" id="businesscontent">
          <h3 style={{ paddingLeft: '20px', color: 'break' }}>Edit my business</h3><br />

          <form className="create-business-form" onSubmit={this.editBusiness}>
            <div className="form-group">
              <label>Name:</label>
              <input type="text" className="field" onChange={this.changelog} name='name' value={this.state.name} required />

            </div><br />
            <div className="form-group">
              <label>Location:</label>
              <input type="text" className="field" onChange={this.changelog} name='location' value={this.state.location} required />
            </div><br />
            <div className="form-group">
              <label>Category:</label>

              <select className="field" name='category' onChange={this.changelog} required>
                <option>{this.state.category}</option>
                <option>Technology</option>
                <option>Health</option>
                <option>Fashion</option>
                <option>Tourism & Hotels</option>
                <option>Education</option>
                <option>Finance & Accounting</option>
                <option>Farming</option>
                <option>Ecommerse</option>
                <option>Real Estate</option>
                <option>Manufacturing</option>
              </select>
            </div><br />

            <div className="form-group">
              <label >About:</label><br /><br />
              <textarea id='textarea' onChange={this.changelog} rows='8' name='about' value={this.state.about} required></textarea>
            </div><br />
            <a style={{ float: 'right', marginTop: '180px' }}><button type="submit" className="btn btn-primary">Update</button></a>
          </form>

        </div>
        <div className="col-md-3"></div>
      </div>
    );
  }
}

EditBusiness.propTypes = {
  params: PropTypes.object.isRequired

};
export default EditBusiness;