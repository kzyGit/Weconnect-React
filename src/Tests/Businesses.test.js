import React from 'react';
import { shallow, mount } from 'enzyme';
import CreateBusiness from '../Components/CreateBusiness';
import EditBusiness from '../Components/EditBusiness';
import Businesses from '../Components/Businesses';
import Business from '../Components/Business';
import Dashboard from '../Components/Dashboard';
import BackNav from '../Components/BackNav';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon'
import moxios from 'moxios';

  
describe('CreateBusinesses component', () => {
    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn()
    };
    global.localStorage = localStorageMock;


    const wrapper = shallow(<CreateBusiness />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('has initial state', () => {
        expect(wrapper.state().businesses).toEqual([]);
    });

    it('handles submit', () => {
        let createBusiness = sinon.spy();
        let wrapper = mount(<CreateBusiness onSubmit={createBusiness} />)
        wrapper.find('form').simulate('submit');
        moxios.wait(() => {
        });
    });
    
    

});

describe('Businesses component', () => {

    const wrapper = shallow(<Businesses />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles searchByName', () => {
        let searchByName = sinon.spy();
        let wrapper = mount(<Businesses onSubmit={searchByName} />)
        wrapper.find('#searchform').simulate('submit');
        moxios.wait(() => {
        });
    });

    it('handles filter', () => {
        let filter = sinon.spy();
        let wrapper = mount(<Businesses onSubmit={filter} />)
        wrapper.find('#filterform').simulate('submit');
        moxios.wait(() => {
        });
    });

    it('handles changename', () => {
        let changename = sinon.spy();
        let wrapper = mount(<Businesses onSubmit={changename} />)
        wrapper.find('#searchname').simulate('change');
        moxios.wait(() => {
        });
    });

    it('handles changefilter', () => {
        let changefilter = sinon.spy();
        let wrapper = mount(<Businesses onSubmit={changefilter} />)
        wrapper.find('#searchfilter').simulate('change');
        moxios.wait(() => {
        });
    });
});

describe('Dashboard component', () => {

    const wrapper = shallow(<Dashboard />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});

describe('BackNav component', () => {
    const wrapper = shallow(<BackNav />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});

describe('EditBusiness component', () => {
    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn()
    };
    global.localStorage = localStorageMock;
    const params = {
        params: {
            id: 1,
        },
    };
    const wrapper = shallow(<EditBusiness params={{ params }} />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    
    
    it('has initial state', () => {
        expect(wrapper.state().id).toEqual('');
        expect(wrapper.state().businesses).toEqual([]);
        expect(wrapper.state().name).toEqual('');
        expect(wrapper.state().location).toEqual('');
        expect(wrapper.state().category).toEqual('');
        expect(wrapper.state().about).toEqual('');
    });

    it('changes state', () => {
        wrapper.setState({ name: 'Andela', location: 'TRM', category: 'tech', about: 'Awesome' });
        expect(wrapper.find('[name="name"]').props().value).toEqual('Andela');
        expect(wrapper.find('[name="location"]').props().value).toEqual('TRM');
        expect(wrapper.find('[name="about"]').props().value).toEqual('Awesome');
    });

    it('has the correct form fields', () => {
        expect(wrapper.find('[name="name"]')).toHaveLength(1);
        expect(wrapper.find('[name="location"]')).toHaveLength(1);
        expect(wrapper.find('[name="about"]')).toHaveLength(1);
        expect(wrapper.find('[name="category"]')).toHaveLength(1);
      });
});   


