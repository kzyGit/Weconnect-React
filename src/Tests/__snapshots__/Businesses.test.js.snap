// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackNav component renders properly 1`] = `
<span
  className="glyphicon glyphicon-circle-arrow-left"
  id="backnav"
  onClick={[Function]}
  style={
    Object {
      "color": "white",
      "cursor": "pointer",
      "fontSize": "22px",
      "paddingLeft": "7%",
    }
  }
/>
`;

exports[`Businesses component renders properly 1`] = `
<div
  className="businessescontent"
>
  <div
    className="row"
  >
    <h3
      onClick={[Function]}
      style={
        Object {
          "color": "maroon",
          "cursor": "pointer",
          "paddingLeft": "20px",
        }
      }
    >
      Businesses
    </h3>
    <br />
    <div
      className="row"
    >
      <div
        className="col-sm-8"
        id="filterspan"
      >
        <form
          id="filterform"
          onSubmit={[Function]}
        >
          <span>
            Category: 
          </span>
          <span>
            <select
              id="search"
              name="searchcategory"
              onChange={[Function]}
            >
              <option
                value=""
              >
                All
              </option>
              <option>
                Technology
              </option>
              <option>
                Tourism & Hotels
              </option>
              <option>
                Health
              </option>
              <option>
                Education
              </option>
              <option>
                Finance & Accounting
              </option>
              <option>
                Farming
              </option>
              <option>
                Ecommerse
              </option>
              <option>
                Real Estate
              </option>
              <option>
                Manufacturing
              </option>
            </select>
          </span>
          <span
            style={
              Object {
                "paddingLeft": "20px",
              }
            }
          >
            Location: 
          </span>
          <span>
            <input
              id="searchfilter"
              name="searchlocation"
              onChange={[Function]}
              type="text"
            />
          </span>
          <span>
            <button
              id="search"
              style={
                Object {
                  "paddingRight": "20px",
                }
              }
              type="submit"
            >
              <span
                className="glyphicon glyphicon-search"
              />
            </button>
          </span>
        </form>
      </div>
      <div
        className="col-sm-1"
      />
      <div
        className="col-sm-3"
      >
        <form
          id="searchform"
          onSubmit={[Function]}
        >
          <span>
            <input
              id="searchname"
              name="searchname"
              onChange={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </span>
          <span>
            <button
              id="search"
              style={
                Object {
                  "paddingRight": "20px",
                }
              }
              type="submit"
            >
              <span
                className="glyphicon glyphicon-search"
              />
            </button>
          </span>
        </form>
      </div>
    </div>
     
    <br />
    <br />
    <div
      className="col-md-12"
    >
      <img
        alt=""
        id="loadpage"
        src="https://upload.wikimedia.org/wikipedia/commons/b/b1/Loading_icon.gif"
      />
    </div>
  </div>
</div>
`;

exports[`CreateBusinesses component renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  />
  <div
    className="col-md-6"
    id="businesscontent"
  >
    <h3>
      Create a business
    </h3>
    <br />
    <form
      className="create-business-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Name:
        </label>
        <input
          className="field"
          name="business_name"
          required={true}
          type="text"
        />
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          Location:
        </label>
        <input
          className="field"
          name="location"
          required={true}
          type="text"
        />
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          Category:
        </label>
        <select
          className="field"
          name="category"
          required={true}
        >
          <option>
            Technology
          </option>
          <option>
            Health
          </option>
          <option>
            Fashion
          </option>
          <option>
            Tourism & Hotels
          </option>
          <option>
            Education
          </option>
          <option>
            Finance & Accounting
          </option>
          <option>
            Farming
          </option>
          <option>
            Ecommerse
          </option>
          <option>
            Real Estate
          </option>
          <option>
            Manufacturing
          </option>
        </select>
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          About:
        </label>
        <br />
        <br />
        <textarea
          id="textarea"
          name="about"
          required={true}
          rows="8"
        />
      </div>
      <a
        style={
          Object {
            "float": "right",
            "marginTop": "180px",
          }
        }
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Create
        </button>
      </a>
    </form>
  </div>
  <div
    className="col-md-3"
  />
</div>
`;

exports[`Dashboard component renders properly 1`] = `
<div
  className="content"
>
  <div
    className="row"
  >
    <h3
      style={
        Object {
          "color": "maroon",
          "paddingLeft": "20px",
        }
      }
    >
      My Businesses
    </h3>
    <br />
    <br />
    <div
      className="col-md-1"
    />
    <div
      className="col-md-10"
    >
      <span
        id="rev"
      />
      <br />
      Let your businesses join the WeConnect network
      <Link
        onlyActiveOnIndex={false}
        style={
          Object {
            "paddingLeft": "30px",
            "textDecoration": "none",
          }
        }
        to="/addbusiness"
      >
        <span
          className="glyphicon glyphicon-hand-right"
        />
         Add Business
      </Link>
      <br />
      <br />
      <br />
    </div>
    <div
      className="col-md-1"
    />
  </div>
</div>
`;

exports[`EditBusiness component renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  />
  <div
    className="col-md-6"
    id="businesscontent"
  >
    <h3
      style={
        Object {
          "color": "break",
          "paddingLeft": "20px",
        }
      }
    >
      Edit my business
    </h3>
    <br />
    <form
      className="create-business-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Name:
        </label>
        <input
          className="field"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          Location:
        </label>
        <input
          className="field"
          name="location"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          Category:
        </label>
        <select
          className="field"
          name="category"
          onChange={[Function]}
          required={true}
        >
          <option />
          <option>
            Technology
          </option>
          <option>
            Health
          </option>
          <option>
            Fashion
          </option>
          <option>
            Tourism & Hotels
          </option>
          <option>
            Education
          </option>
          <option>
            Finance & Accounting
          </option>
          <option>
            Farming
          </option>
          <option>
            Ecommerse
          </option>
          <option>
            Real Estate
          </option>
          <option>
            Manufacturing
          </option>
        </select>
      </div>
      <br />
      <div
        className="form-group"
      >
        <label>
          About:
        </label>
        <br />
        <br />
        <textarea
          id="textarea"
          name="about"
          onChange={[Function]}
          required={true}
          rows="8"
          value=""
        />
      </div>
      <br />
      <a
        style={
          Object {
            "float": "right",
            "marginTop": "180px",
          }
        }
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Update
        </button>
      </a>
    </form>
  </div>
  <div
    className="col-md-3"
  />
</div>
`;
